src/sortStudents.ts

```diff
@@ -2,6 +2,94 @@
 // create and export SortType enum
 // create SortOrder type
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+type SortOrder = 'asc' | 'desc';
+
+export enum SortType {
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
+}
+
+type Student = {
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
+};
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const sortedArray: Student[] = [];
+  const sum = (prev: number, curr: number): number => {
+    return prev + curr;
+  };
+
+  students.forEach((student: Student) => {
+    sortedArray.push({ ...student });
+    sortedArray[sortedArray.length - 1].grades = [...student.grades];
+  });
+
```

Comment:
For copy students you can use map method:
```suggestion
  const sortedArray: Student[] = students.map((student: Student) => (
    ({ ...student })
  ));

```